lib/services/database.go

```diff
@@ -133,6 +133,13 @@ func NewDatabaseFromRDSInstance(instance *rds.DBInstance) (types.Database, error
 
 // NewDatabaseFromRDSCluster creates a database resource from an RDS cluster (Aurora).
 func NewDatabaseFromRDSCluster(cluster *rds.DBCluster) (types.Database, error) {
+	// Aurora Serverless (v1 and v2) does not support IAM authentication
+	// https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/aurora-serverless.html#aurora-serverless.limitations
+	// https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/aurora-serverless-2.limitations.html
+	if aws.StringValue(cluster.EngineMode) == RDSEngineModeServerless {
```

Comment:
Also, what about other engine modes you mentioned in the description like `provisioned`, `parallelquery`, `global`, `multimaster`? Do you know if all those support IAM auth or should we skip some of those as well?

Reply:
Good catch.

As for whether they support IAM, it's gets complicated. for example:

> Aurora MySQL 1.22.2, 1.20.1, 1.19.6, and 5.6.10a **_only_**: Parallel query doesn't support AWS Identity and Access Management (IAM) database authentication.

I will convert this check to a function something like `isIAMAuthSupported` and compile all these scenarios.

Reply:
@greedy52 `IAMDatabaseAuthenticationEnabled` indicates whether IAM auth is enabled or disabled on the instance. We use it to determine whether we should attempt to enable IAM on the instance ourselves. So it can't be used to determine whether the instance _supports_ IAM or not.

Reply:
> @greedy52 `IAMDatabaseAuthenticationEnabled` indicates whether IAM auth is enabled or disabled on the instance. We use it to determine whether we should attempt to enable IAM on the instance ourselves. So it can't be used to determine whether the instance _supports_ IAM or not.

That's correct. I guess this is a different topic than the original ticket. So wouldn't `tsh db connect" would also fail if databases that DOES NOT have IAM auth enabled. That said, registering them but direct users to enable iam auth probably better option. 

Will try to capture all the 'not-supported' cases in this pr. 😄